src/main/java/org/cbio/gdcpipeline/writer/ClinicalPatientDataWriter.java

```diff
@@ -0,0 +1,179 @@
+package org.cbio.gdcpipeline.writer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.cbio.CBioClinicalDataModel;
+import org.cbio.gdcpipeline.model.cbio.PatientFileModel;
+import org.cbio.gdcpipeline.util.ClinicalPatientFileHeaders;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.FlatFileHeaderCallback;
+import org.springframework.batch.item.file.FlatFileItemWriter;
+import org.springframework.batch.item.file.ResourceAwareItemWriterItemStream;
+import org.springframework.batch.item.file.transform.BeanWrapperFieldExtractor;
+import org.springframework.batch.item.file.transform.DelimitedLineAggregator;
+import org.springframework.batch.item.file.transform.FieldExtractor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Dixit on 14/06/17.
+ */
+public class ClinicalPatientDataWriter implements ResourceAwareItemWriterItemStream<CBioClinicalDataModel> {
+
+    @Value("#{jobParameters[sourceDirectory]}")
+    private String sourceDir;
+
+    @Value("#{jobParameters[outputDirectory]}")
+    private String outputDir;
+
+    @Value("#{jobParameters[study]}")
+    private String cancer_study_id;
+
+    private String dataOutputFile;
+
+    private Resource resource;
+
+
+    @Value("${clinical.data.patient.file.name}")
+    private String CLINICAL_DATA_PATIENT_FILE_NAME;
+
+    @Value("${clinical.metadata.patient.file.name}")
+    private String CLINICAL_METADATA_PATIENT_FILE_NAME;
+
+    FlatFileItemWriter<PatientFileModel> dataWriter = new FlatFileItemWriter<>();
+
+    @Autowired
+    ClinicalPatientFileHeaders patientFileHeaders;
+
+
+    private static Log LOG = LogFactory.getLog(ClinicalPatientDataWriter.class);
+
+    private String modifyRow(String[] row) {
+        // Tab Delimited
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < row.length; i++) {
+            if (i == 0) {
+                sb.append(row[i]);
+            } else {
+                sb.append("\t");
+                sb.append(row[i]);
+            }
+        }
+        return sb.toString();
+    }
+
+
+    private FlatFileHeaderCallback clinicalDataHeader() {
+        return new FlatFileHeaderCallback() {
+            @Override
+            public void writeHeader(Writer writer) throws IOException {
```

Comment:
Instead of using a stringbuilder as below you should use a metadataManager interface that returns a map, similar to https://github.com/knowledgesystems/cmo-pipelines/blob/master/redcap/redcap_source/src/main/java/org/mskcc/cmo/ks/redcap/source/MetadataManager.java